### PR TITLE
Move verification down

### PR DESCRIPTION
**Pull request by jonmv**
@bratseth please review.

### CODE REVIEW

**Review by bratseth — APPROVED** [review]

**Comment by bratseth** [comment]
```
Failures: 
00:09:06 [ERROR]   FileDBRegistryTestCase.importAndExport:51 expected:<[/files/no_foo.json is not relative]> but was:<[java.io.FileNotFoundException: src/test/apps/zkapp/files/no_foo.json (No such file or directory)]>

```

Keep that check for a nicer error message? Otherwise, why not use the vespajlib path, where you just need to do new Path(...).

**Comment by jonmv** [comment]
PTAL.

File: configserver/src/main/java/com/yahoo/vespa/config/server/filedistribution/ApplicationFileManager.java [hunk 1]
```diff
@@ -31,53 +31,43 @@
...
     }
 
     @Override
-    public FileReference addUri(String uri, String relativePath) {
-        File file = download(uri, relativePath);
-        try {
-            return addFile(file);
-        } catch (IOException e) {
+    public FileReference addUri(String uri, Path path) {
+        throw new UnsupportedOperationException("URI type is not supported");
```
Comment (bratseth) [inline configserver/src/main/java/com/yahoo/vespa/config/server/filedistribution/ApplicationFileManager.java:45]: But we already support this?

**Review by bratseth — CHANGES_REQUESTED** [review]

File: config-model/src/main/java/com/yahoo/searchdefinition/RankProfile.java [hunk 2]
```diff
@@ -882,10 +883,10 @@ private Reader openRankingExpressionReader(String expName, String expression) {
...
 
         String fileName = extractFileName(expression);
-        File file = new File(fileName);
-        if (!file.isAbsolute() && file.getPath().contains("/")) // See ticket 4102122
-            throw new IllegalArgumentException("In " + name() + ", " + expName + ", ranking references file '" + file +
-                                               "' in subdirectory, which is not supported.");
+        Path.fromString(fileName); // No ".."
+        if (fileName.contains("/")) // See ticket 4102122
+            throw new IllegalArgumentException("In " + name() + ", " + expName + ", ranking references file '" +
+                                               fileName + "' in a different directory, which is not supported.");
```
[the remark this answers is not in the document]
Comment (jonmv) [reply config-model/src/main/java/com/yahoo/searchdefinition/RankProfile.java:889]: No, but here we permit `/` if the path is absolute, but then we deny absolute paths further down, which is just nonsense.

File: configserver/src/main/java/com/yahoo/vespa/config/server/filedistribution/ApplicationFileManager.java (hunk 1 above)
Comment (jonmv) [reply configserver/src/main/java/com/yahoo/vespa/config/server/filedistribution/ApplicationFileManager.java:45]: No? The only thing I can find that calls this code is ONNX-models, but they also explicitly throw if the type is URI ...

File: config-model/src/main/java/com/yahoo/searchdefinition/RankProfile.java [hunk 3]
```diff
@@ -882,10 +883,10 @@ private Reader openRankingExpressionReader(String expName, String expression) {
         if (!expression.startsWith("file:")) return new StringReader(expression);
 
         String fileName = extractFileName(expression);
-        File file = new File(fileName);
-        if (!file.isAbsolute() && file.getPath().contains("/")) // See ticket 4102122
-            throw new IllegalArgumentException("In " + name() + ", " + expName + ", ranking references file '" + file +
-                                               "' in subdirectory, which is not supported.");
+        Path.fromString(fileName); // No ".."
```
[the remark this answers is not in the document]
Comment (jonmv) [reply config-model/src/main/java/com/yahoo/searchdefinition/RankProfile.java:886]: No: 
```java

        System.err.println(Path.fromString("").elements());
        System.err.println(Path.fromString("a").elements());
        System.err.println(Path.fromString("/").elements());
        System.err.println(Path.fromString("a/").elements());
        System.err.println(Path.fromString("/a").elements());
        ```
results in
```
[]
[a]
[]
[a]
[a]
```

File: configserver/src/main/java/com/yahoo/vespa/config/server/filedistribution/ApplicationFileManager.java (hunk 1 above)
Comment (bratseth) [reply configserver/src/main/java/com/yahoo/vespa/config/server/filedistribution/ApplicationFileManager.java:45]: Ok!

File: config-model/src/main/java/com/yahoo/searchdefinition/RankProfile.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (bratseth) [reply config-model/src/main/java/com/yahoo/searchdefinition/RankProfile.java:886]: Yes, all those with size()==1 would be interpreted as the file "a" which is safe, but ok.

File: config-model/src/main/java/com/yahoo/searchdefinition/RankProfile.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (bratseth) [reply config-model/src/main/java/com/yahoo/searchdefinition/RankProfile.java:889]: Yes, absolute are fine, but what about "myExpressions/expr1.expression"?

[the remark this answers is not in the document]
Comment (bratseth) [reply config-model/src/main/java/com/yahoo/searchdefinition/RankProfile.java:889]: I *think* it's not supported but one of us should be sure they verified it ...

[the remark this answers is not in the document]
Comment (bratseth) [reply config-model/src/main/java/com/yahoo/searchdefinition/RankProfile.java:889]: It's better to support as little as possible :-)

File: configserver/src/main/java/com/yahoo/vespa/config/server/filedistribution/ApplicationFileManager.java (hunk 1 above)
Comment (jonmv) [reply configserver/src/main/java/com/yahoo/vespa/config/server/filedistribution/ApplicationFileManager.java:45]: I was wrong 😭

**Pull request merged**